### ISSUE TITLE
git-sync: ask what to do when a remote branch doesn't exist

### ISSUE THREAD

**Issue by kevgo** [Originate/git-town#663]
situation
- I have a feature branch B that is a child of feature branch A
- on another machine I ship branch A
- on my initial machine I run `git sync`

expected behavior
- GT tells me that branch A seems gone
- GT asks me whether I want to push it back up or remove it from my machine

current behavior
- GT always pushes branch A back to the repo, even though it is no longer needed

This requires a change in behavior:
- `git hack` always pushes the branch to the remote
- `git sync` assumes that branches always have a remote branch. If it isn't there, it asks whether to delete the branch, or sync it back

**Comment by kevgo** [comment]
blocked by #664
